bitcoin/src/blockdata/transaction.rs

```diff
@@ -1278,11 +1278,12 @@ pub struct InputWeightPrediction {
 
 impl InputWeightPrediction {
     /// Computes the prediction for a single input.
-    pub fn new<I>(input_script_len: usize, witness_element_lengths: I) -> Self
-        where I: IntoIterator<Item = usize>,
+    pub fn new<T>(input_script_len: usize, witness_element_lengths: T) -> Self
+        where T :IntoIterator, T::Item:Borrow<usize>,
```

Comment:
I didn't mean to rename the type parameter, I just misremembered it. :) It doesn't matter anyway.